api/envoy/config/route/v3/route_components.proto

```diff
@@ -1099,6 +1119,15 @@ message RouteAction {
   // :ref:`HTTP_DOWNSTREAM_STREAM_IDLE <envoy_v3_api_enum_value_config.overload.v3.ScaleTimersOverloadActionConfig.TimerType.HTTP_DOWNSTREAM_STREAM_IDLE>`.
   google.protobuf.Duration idle_timeout = 24;
 
+  // If absent, allows safe requests by default.
```

Comment:
Is this for downstream requests or upstream requests?  Please clarify in the comment.

If this is for upstream requests, should it be part of the cluster definition instead of the route?

Reply:
Router filter takes in various configs: the http filter config, cluster info and route. I was also wondering which config these should go. @alyssawilk thoughts on this?

Reply:
One reason I put it here is that this config is tightly associated with retry_policy. If one configures a POST request to be able to use early data, they would better also adjust the retry policy to that cluster.

Reply:
yeah I think having it associated with retry policy makes sense